packages/cli-kit/src/private/node/ui/components/TextPrompt.tsx

```diff
@@ -25,6 +30,10 @@ export interface TextPromptProps {
 
 const TextPrompt: FunctionComponent<TextPromptProps> = ({
   message,
+  infoTable,
+  finalInstruction,
+  exitOnEsc = false,
```

Comment:
I wonder if this behavior can be applied to all prompts? What was the reason for introducing it only for dangerous confirmation prompts?

Reply:
It definitely makes sense to use a common pattern for this. However, text prompts are unusual in that it's hard to provide a "cancel" option when all typing just goes to the text field. So I think it makes sense to start here and wait for a known use case for another prompt.

Currently we assume that all other prompts will complete successfully, or the process will be killed. I think that makes downstream code easier to write clearly.

In the unique case of a dangerous confirmation prompt, what we're promising is either `true` or `false` as a return value, so usual boolean handling rules will apply. And this is the same as the behavior of a non-dangerous confirmation prompt.

In contrast, our other prompts might return a `string` or other type, and `string or undefined` gets messy when you're in the middle of a flow. Sticking with an explicit error might be best.

Reply:
The non-dangerous confirmation prompt gives you the ability to exit by selecting the `no` option. But there's no such option here, so we substitute with `esc`.